### PR TITLE
Drakon updates 2

### PR DESCRIPTION
**Pull request by JagdPietr**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by JagdPietr** [comment]
Updates in this PR:
1. Transfer of drakon to jagdpietr dir
2. Fix of encoder switch placement and orientation
3. Transfer of OLED stuff from keymap(s).c to keyboard.c for OLED functionality

File: keyboards/jagdpietr/drakon/drakon.c [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+        if (clockwise) {
+            tap_code(KC_VOLU);
+        } else {
+            tap_code(KC_VOLD);
+        }
+}
+
+#ifdef OLED_DRIVER_ENABLE
+
+__attribute__ ((weak))
```
Comment (drashna) [inline keyboards/jagdpietr/drakon/drakon.c:31]:
```suggestion
```

File: keyboards/jagdpietr/drakon/drakon.c [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+enum Layer_names {
+_Base,
+_FN,
+_Lyr2
+};
+
+oled_rotation_t oled_init_user(oled_rotation_t rotation) {
```
Comment (drashna) [inline keyboards/jagdpietr/drakon/drakon.c:38]:
```suggestion
__attribute__((weak))
oled_rotation_t oled_init_user(oled_rotation_t rotation) {
```

File: keyboards/jagdpietr/drakon/drakon.c [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+    return OLED_ROTATION_90;  // flips the display 90 degrees if offhand
+
+}
+
+void render_status(void) {
```
Comment (drashna) [inline keyboards/jagdpietr/drakon/drakon.c:43]:
```suggestion
static void render_status(void) {
```

File: keyboards/jagdpietr/drakon/drakon.c [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+            oled_off();
+        } else {
+            if(timer_elapsed32(anim_timer) > ANIM_FRAME_DURATION) {
+                anim_timer = timer_read32();
+                animation_phase();
+            }
+        }
+    }
+}
+void oled_task_user(void) {
```
Comment (drashna) [inline keyboards/jagdpietr/drakon/drakon.c:210]:
```suggestion
__attribute__((weak))
void oled_task_user(void) {
```

**Comment by JagdPietr** [comment]
> This branch needs to be rebased upon QMK's current `master` branch. As you've already added this board in #10071, what this PR would do when merged is add a duplicate under a new folder, instead of moving the existing files (modified or not) to the new folder.
> 
> To sort this properly, you should [resynchronize your master branch](https://docs.qmk.fm/#/newbs_git_resynchronize_a_branch), then run the following commands:
> 
> ```shell
> # check out this branch:
> git checkout drakon-updates-2
> 
> # rewrite this branch to match the now-synchronized master:
> git reset --hard master
> 
> # create the jagdpietr directory, which won't exist at this point:
> mkdir -p keyboards/jagdpietr/
> 
> # move the keyboard source files:
> git mv -- keyboards/drakon/ keyboards/jagdpietr/drakon/
> 
> # remove the logo keymap, which I notice isn't present here:
> git rm -rf -- keyboards/jagdpietr/drakon/keymaps/logo/
> 
> # commit the changes:
> git commit -m "move Drakon to jagdpietr directory"
> 
> # update this pull request:
> git push --force-with-lease
> ```

noted and finished the instructions
thank you very much!

**Comment by noroadsleft** [comment]
@JagdPietr Your resynchronization wasn't done properly; there should only be one commit on this PR.

**Comment by JagdPietr** [comment]
I apologize for that; will redo this action

**Comment by JagdPietr** [comment]
Finished with the instructions already, and I hope this is correct this time.
However, I'm quite worried because the changes on keyboard.c and keymap/s.c were not reflected on this latest commit when I viewed them.

**Comment by drashna** [comment]
Yeah, the changes don't seem to be reflected, unfortunately. :(

**Review by noroadsleft — CHANGES_REQUESTED** [review]
Also, the changes to `lib/vusb/` need to be reverted:

From your `qmk_firmware` directory, run:

```shell
cd lib/vusb/
git checkout bdb53e4c043d089279d9891b68bea77614cb97ee
cd ../../
git add -- lib/vusb/
git commit -m "fix vusb submodule"
git push
```

**Comment by JagdPietr** [comment]
Changes and recommendations finished - I just hope I got them all correct :)

**Review by noroadsleft — APPROVED** [review]

**Comment by JagdPietr** [comment]
Hi! May I ask what should I do to finish this PR? It has taken a bit while now, and maybe I missed something to do to finish this PR.

**Review by fauxpark — APPROVED** [review]

**Comment by noroadsleft** [comment]
I don't recall if it was asked, but are there any of these in the wild?

**Comment by JagdPietr** [comment]
I have twp working prototypes in my care, and another in the proxy. The GB is still on production phase.

**Comment by noroadsleft** [comment]
Ah, wanted to make sure we're not messing things up for end users if we move this now.

**Comment by noroadsleft** [comment]
Thanks!

**Comment by JagdPietr** [comment]
Thank you very much for all of you! Such a great team :)

**Pull request merged**